app/cmake/ZephyrBuildConfig.cmake

```diff
@@ -74,34 +97,58 @@ foreach(root ${BOARD_ROOT})
 	endif()
 
 	if(DEFINED SHIELD)
-		find_path(shields_refs_list
-		    NAMES ${SHIELD}.overlay
-		    PATHS ${root}/boards/shields/*
-		    NO_DEFAULT_PATH
-		    )
-		foreach(shield_path ${shields_refs_list})
-			get_filename_component(SHIELD_DIR ${shield_path} NAME)
-			list(APPEND KEYMAP_DIRS ${shield_path})
+		foreach(s ${SHIELD_AS_LIST})
+			if(NOT ${s} IN_LIST SHIELD_LIST)
+				message(WARNING "Didn't find ${s}")
+				continue()
+			endif()
+			message(STATUS "Adding ${SHIELD_DIR_${s}}")
+			list(APPEND KEYMAP_DIRS ${SHIELD_DIR_${s}})
+			get_filename_component(shield_dir_name ${SHIELD_DIR_${s}} NAME)
+			list(APPEND SHIELD_DIR ${shield_dir_name})
 		endforeach()
 	endif()
 endforeach()
 
+# Give a shield like `kyria_rev2_left` we want to use `kyria_rev2` and `kyria` as candidate names for
+# overlay/conf/keymap files.
+if(DEFINED SHIELD)
+	foreach(s ${SHIELD_AS_LIST})
+		message(STATUS ${SHIELD_DIR_${s}})
+		get_filename_component(shield_dir_name ${SHIELD_DIR_${s}} NAME)
+		string(REPLACE "_" ";" S_PIECES ${s})
+		list(LENGTH S_PIECES S_PIECES_LEN)
+		foreach (n RANGE ${S_PIECES_LEN-2})
```

Comment:
The -2 here was a bit confusing until I looked up that cmake loops to the stop index inclusive.

Since cmake lists are just semicolon-separated strings, would this work and be easier to understand?

```cmake
while (NOT S_PIECES STREQUAL "")
```

(I assume that `list(POP_BACK S_PIECES)` would eventually result in S_PIECES being an empty string, so the loop would be guaranteed to terminate, but I am not very familiar with cmake.)

Reply:
Much nicer, indeed.